From 9810f6d21a0c31b18ac38c56da27f592f245e686 Mon Sep 17 00:00:00 2001
Message-Id: <patch-3.5-9810f6d21a-20210714T000248Z-avarab@gmail.com>
In-Reply-To: <cover-0.5-0000000000-20210714T000248Z-avarab@gmail.com>
References: <87czsv2idy.fsf@evledraar.gmail.com>
	<cover-0.5-0000000000-20210714T000248Z-avarab@gmail.com>
From: =?UTF-8?q?=C3=86var=20Arnfj=C3=B6r=C3=B0=20Bjarmason?=
 <avarab@gmail.com>
Date: Tue, 13 Jul 2021 15:18:44 +0200
Subject: [PATCH 3/5] SANITIZE tests: fix memory leaks in t0500*, add to
 whitelist
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Fix a couple of memory leaks in running t0500* and remove the
SANITIZE_LEAK prerequisite on the one test we were skipping.

The trace2 issue is a memory leak introduced in ee4512ed48 (trace2:
create new combined trace facility, 2019-02-22), that was seemingly
done by omission. We already had a "unset_self" function to free the
resources, let's free the thread name there.

The other issue was a leak in the progress.c test helper itself, added
in 2bb74b53a4 (Test the progress display, 2019-09-16).

Signed-off-by: Ævar Arnfjörð Bjarmason <avarab@gmail.com>
---
 t/helper/test-progress.c    | 1 +
 t/t0500-progress-display.sh | 2 +-
 trace2/tr2_tls.c            | 1 +
 3 files changed, 3 insertions(+), 1 deletion(-)

diff --git a/t/helper/test-progress.c b/t/helper/test-progress.c
index 5d05cbe789..9265e6ab7c 100644
--- a/t/helper/test-progress.c
+++ b/t/helper/test-progress.c
@@ -69,6 +69,7 @@ int cmd__progress(int argc, const char **argv)
 			die("invalid input: '%s'\n", line.buf);
 	}
 	stop_progress(&progress);
+	strbuf_release(&line);
 
 	return 0;
 }
diff --git a/t/t0500-progress-display.sh b/t/t0500-progress-display.sh
index 7afb9abb1f..a56ade16b2 100755
--- a/t/t0500-progress-display.sh
+++ b/t/t0500-progress-display.sh
@@ -284,7 +284,7 @@ test_expect_success 'cover up after throughput shortens a lot' '
 	test_cmp expect out
 '
 
-test_expect_success !SANITIZE_LEAK 'progress generates traces' '
+test_expect_success 'progress generates traces' '
 	cat >in <<-\EOF &&
 	throughput 102400 1000
 	update
diff --git a/trace2/tr2_tls.c b/trace2/tr2_tls.c
index 067c23755f..f55827e24e 100644
--- a/trace2/tr2_tls.c
+++ b/trace2/tr2_tls.c
@@ -96,6 +96,7 @@ void tr2tls_unset_self(void)
 	pthread_setspecific(tr2tls_key, NULL);
 
 	free(ctx->array_us_start);
+	strbuf_release(&ctx->thread_name);
 	free(ctx);
 }
 
-- 
2.32.0-dev

